frontend/www/js/omegaup/arena/problemStore.ts

```diff
@@ -1,19 +1,44 @@
 import Vue from 'vue';
 import Vuex from 'vuex';
+import { types } from '../api_types';
 
 Vue.use(Vuex);
 
-interface ProblemState {
-  showOverlay: boolean;
+export interface ProblemState {
+  // The mapping of problem alias to indices on the problems array
+  problems: types.ProblemInfo[];
+
+  index: Record<string, number>;
 }
 
-const problemStore = new Vuex.Store<ProblemState>({
-  state: { showOverlay: false },
+const problemsStore = new Vuex.Store<ProblemState>({
+  state: {
+    problems: [],
+    index: {},
+  },
   mutations: {
-    toggleOverlay(state, showOverlay: boolean) {
-      state.showOverlay = !showOverlay;
+    addProblem(state, problem: types.ProblemInfo) {
+      if (Object.prototype.hasOwnProperty.call(state.index, problem.alias)) {
+        Vue.set(
+          state.problems,
+          state.index[problem.alias],
+          Object.assign(
+            {},
+            state.problems[state.index[problem.alias]],
+            problem,
+          ),
+        );
+        return;
+      }
+      Vue.set(state.index, problem.alias, state.problems.length);
+      state.problems.push(problem);
+    },
+    // Just in case be necessary delete the list of problems
```

Comment:
```suggestion
    // Just in case it's necessary to delete the list of problems
```

aunque eso suena YAGNIoso. qué tal si se elimina este código en caso de que _no_ sea necesario eliminar problemas?

Reply:
Se va por lo pronto esto. 

No conocía el acrónimo, tuve que googlearlo xD